portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,28 +14,108 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.FetchOptions.Builder;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import com.google.gson.Gson;
+import com.google.sps.data.Comment;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList; 
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+/** Servlet responsible for handling comments data */
+@WebServlet("/comment")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> messages = new ArrayList<String>();
-    messages.add("Hello");
-    messages.add("Testing JSON");
-    messages.add("I hope this works!");
+    String sort = request.getParameter("sort");
+
+    // Sort comments according to request parameter
+    Query query;
+    FilterPredicate filter = new FilterPredicate("img", Query.FilterOperator.EQUAL, request.getParameter("img"));
+    if (sort.equals("new")) {
+      query = new Query("Comment").setFilter(filter).addSort("timestamp", SortDirection.DESCENDING);
+    } else {
+      query = new Query("Comment").setFilter(filter).addSort("timestamp", SortDirection.ASCENDING);
+    }
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    int numComments = results.countEntities(FetchOptions.Builder.withDefaults());
+
+    // Set the number of comments to print (max)
+    String quantity = request.getParameter("quantity");
+    int max = -1;
+    if (quantity != null && quantity.length() > 0) {
+      max = Integer.parseInt(quantity);
+    } else {
+      max = numComments;
+    }
+
+    // Create Comment objects and add to comments ArrayList
+    ArrayList<Comment> comments = new ArrayList<Comment>();
+    for (Entity entity : results.asIterable(FetchOptions.Builder.withLimit(max))) {
+      long id = entity.getKey().getId();  
+      String comment = (String) entity.getProperty("comment");
```

Comment:
Just a nit: when you initialize you object, comment is after name, but u get the comment before name (switch this line and next line)